packages/mattermost-redux/src/actions/search.ts

```diff
@@ -299,6 +299,31 @@ export function removeSearchTerms(teamId: string, terms: string): ActionFunc {
     };
 }
 
+export function getRecentSearches(): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
```

Comment:
You can actually replace this with the `bindClientFunc` helper function which covers all the try/catch and error handling logic. You just need to give it the `Client4` method to call and the `SearchTypes.RECENT_SEARCHES_SUCCESS` to dispatch on success, and that'll be it

Reply:
That's way simpler, thanks!